llvm/include/llvm/CodeGen/GlobalISel/Utils.h

```diff
@@ -343,6 +343,8 @@ inline bool isKnownNeverSNaN(Register Val, const MachineRegisterInfo &MRI) {
   return isKnownNeverNaN(Val, MRI, true);
 }
 
+bool isKnownNeverZeroFloat(Register Val, const MachineRegisterInfo &MRI);
```

Comment:
For this, I was planning on renaming GISelKnownBits to GISelValueTracking, and just start with the generalized computeKnownFPClass. The DAG has much weaker versions of all of these functions, we're better off copying the IR pattern